trino/trino.go

```diff
@@ -976,11 +1022,34 @@ func (qr *driverRows) fetch(allowEOF bool) error {
 	qr.rowindex = 0
 	qr.data = qresp.Data
 	qr.nextURI = qresp.NextURI
+
+	qrStats := QueryProgressCallbackResponse{
+		ID:         qresp.ID,
+		QueryStats: qresp.Stats,
+		Error:      qresp.Error,
+	}
+
 	if len(qr.data) == 0 {
 		if qr.nextURI != "" {
+			if qr.stmt.conn.callback != nil {
```

Comment:
The body of this condition should be a separate method.

Reply:
Why this is only called when there's no data?

Reply:
It maybe due to my lack of understanding trino rest api and go code, 
but I thought that running state is when data is being processed and not available to user yet.

Reply:
That's an implementation detail. This should be always called when there's a response from the server.

Reply:
As of now, I changed so that changed so that updates when there is change in any state OR if callback period has passed. :)
```go
func (qr *driverRows) updateCallback(qrStats QueryProgressCallbackResponse) {
	currentTime := time.Now()
	diff := currentTime.Sub(qr.stmt.conn.callbackPeriod.LastCallbackTime)
	period := qr.stmt.conn.callbackPeriod.Period
	// Check if period has passed OR if query state changes
	if diff >= period || qrStats.LastQueryState != qrStats.QueryStats.State {
		qr.stmt.conn.callback.Update(qrStats)
		qr.stmt.conn.callbackPeriod.LastCallbackTime = currentTime
	}
}
```

Reply:
BTW am working on change this to channel (am keep getting interrupts so it is taking some time) i'll let you know when am done with it